src/DynamicData/Internal/Rxx.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+#if NET9_0_OR_GREATER
+
+namespace System.Reactive.Linq;
+
+internal static class Rxx
+{
+    public static IObservable<T> Synchronize<T>(this IObservable<T> source, Lock locker) =>
+        new Synchronize<T>(source, locker).Run();
+}
+
+/// <summary>
+/// Keep this class internal as it should be supplied by System.Reactive and probably will be one day.
+/// </summary>
+internal sealed class Synchronize<T>
+{
+    private readonly IObservable<T> _source;
+    private readonly Lock _locker;
```

Comment:
Recommend the name `_gate`, that's generally the terminology Microsoft and RX use for objects used to power a lock.

Reply:
I'll stick to locker as it's ubiquitous throughout the code base.